sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3718,6 +3718,49 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-32976: Support column list specification in insert into statement") {
```

Comment:
can we create a `insert.sql` test file and move this test there?

Reply:
Based on this requirement, I create the base trait and 3 concrete test suites, but w/ `V1SQLInsertTestSuite` trait for bypassing different error handling between v2/v1 tables, which I guess is not the right thing to do in this PR.

FYI, when the final query's output mismatches w/ the target table output, v1 throws errors at org.apache.spark.sql.execution.datasources.DataSourceAnalysis#convertStaticPartitions, while v2 at org.apache.spark.sql.catalyst.analysis.TableOutputResolver#resolveOutputColumns

To merge these two parts rashly would introduce a large number of LOC in other test cases and may
 make this PR hard to review.